### DISCUSSION TITLE
Immediate shovel execution using API

### DISCUSSION THREAD

**Discussion by amigian74** [rabbitmq/rabbitmq-server#4326] [Questions] [answered]
Hi,

I'm trying to create a dynamic shovel, which will shovel all messages immediately after creation. The shovel is created as excpected but nothing happens. Is this even possible (it is with the Management Console)?

Thx in advance Steven

**Comment by lukebakken** [comment]
Yes, this should work. We need more information to assist:
* RabbitMQ and Erlang version
* A more detailed description of how you're creating the shovel. What values are you using?
* Export of your definitions before creating the shovel. Please *ATTACH* the file to your response.
* Export after creating the shovel.

**Comment by amigian74** [comment, marked as answer]
Thx for the reply, but we found out the problem. I did not specifiy a user for the shovel and the default user was not existing. With the right user given it works.

Thx a lot anyway
